### PR TITLE
Add Delete Evaluation Endpoint

### PR DESCRIPTION
**Pull request by czaloom**
Changes
- Added endpoint: `DELETE /evaluations/{evaluation_id}`

### CODE REVIEW

File: api/valor_api/backend/core/evaluation.py [hunk 1]
```diff
@@ -1055,3 +1055,34 @@ def delete_evaluations(
...
+    if evaluation.status in {
+        enums.EvaluationStatus.PENDING,
+        enums.EvaluationStatus.RUNNING,
+    }:
+        raise exceptions.EvaluationRunningError
+    elif evaluation.status in enums.EvaluationStatus.DELETING:
+        return
+
+    try:
+        db.delete(evaluation)
```
Comment (ekorman) [inline api/valor_api/backend/core/evaluation.py:1084]: is this automatically going to delete metrics associated to the evaluation?

File: api/valor_api/backend/core/evaluation.py [hunk 2]
```diff
@@ -1055,3 +1055,34 @@ def delete_evaluations(
...
+    Delete a evaluation by id.
+
+    Parameters
+    ----------
+    db : Session
+        The database session.
+    evaluation_id : int
+        The evaluation identifer.
+
+    Raises
```
Comment (ntlind) [inline api/valor_api/backend/core/evaluation.py:1071]: remove raise doc?

File: api/valor_api/backend/core/evaluation.py (hunk 1 above)
Comment (czaloom) [reply api/valor_api/backend/core/evaluation.py:1084]: Yes, sqlalchemy ensures that metrics will be deleted.

```
settings: Mapped[Evaluation] = relationship(back_populates="metrics")
```

Comment (czaloom) [reply api/valor_api/backend/core/evaluation.py:1084]: We delete evaluations in a similar manner when a model or dataset is deleted.

File: api/valor_api/backend/core/evaluation.py (hunk 2 above)
Comment (czaloom) [reply api/valor_api/backend/core/evaluation.py:1071]: Added it in!

**Review by ekorman — APPROVED** [review]

**Pull request merged**